### ISSUE TITLE
Probability and sigmoid function

### ISSUE THREAD

**Issue by kennethchoe** [ageron/handson-ml#103]
I just finished chapter 4. When we say _probability_ from softmax_reg, do we mean the same _probability_ as in "the probability of tossing a coin and getting head is 50%"?
It seems to me that sigmoid function is great way to translate values into something close to _probability_, but I'm not sure how it can be exactly _probability_ itself. Then again, I'm not sure how softmax_reg mapping of these individual _probability_ can lead to proper _probability_ of each classification. So I wonder if both sigmoid and softmax are really a way to represent _approximate_ _probability_ that can be easily calculated mathematically.

Really great book by the way.

**Comment by ageron** [comment]
Great question, and thanks for the kind words.  The output of a neural network that uses a sigmoid or softmax activation function at its output is usually interpreted as an "estimated probability".
I tried to write "estimated probability" rather than just "probability" as often as I could in the book, but perhaps I sometimes forgot and just wrote "probability" (I'll double check).  I wrote a few more details about how to interpret the output of a classification model in a similar question (issue #91).
Hope this helps. :)

**Comment by ageron** [comment]
Note: many people prefer to say "confidence" rather than "estimated probability".  When someone says "I'm 70% confident" about something, it's clear that this number is meant to approximate a probability, but it's also clear that it may be completely off.  It may also be consistently off: for example, I've noticed that when people say "I'm 99% confident", they are right about 75% of the time.  At least, I'm 99% confident about that.  ;-)

**Comment by kennethchoe** [comment]
Best answer ever.
